sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -3170,22 +3172,24 @@ abstract class CSVSuite
     }
   }
 
-  test("SPARK-42237: change binary to unsupported dataType") {
-    withTempPath { path =>
-      val colName: String = "value"
-      checkError(
-        exception = intercept[AnalysisException] {
-          Seq(Array[Byte](1, 2))
-            .toDF(colName)
-            .write
-            .csv(path.getCanonicalPath)
-        },
-        errorClass = "UNSUPPORTED_DATA_TYPE_FOR_DATASOURCE",
-        parameters = Map(
-          "columnName" -> s"`$colName`",
-          "columnType" -> "\"BINARY\"",
-          "format" -> "CSV")
-      )
+  test("SPARK-48807: Binary support for csv") {
+    BinaryOutputStyle.values.foreach { style =>
+      withTempPath { path =>
+        withSQLConf(SQLConf.BINARY_OUTPUT_STYLE.key -> style.toString) {
+          val df = Seq((1, "Spark SQL".getBytes())).toDF("id", "value")
+          df.write
+            .option("ds_option", "value")
+            .format(dataSourceFormat)
+            .save(path.getCanonicalPath)
+          val expectedStr = ToStringBase.getBinaryFormatter("Spark SQL".getBytes())
```

Comment:
Can we change the value as non UTF8 output instead?

Reply:
This helper method gets a binary formatter based on BINARY_OUTPUT_STYLE and converts the raw bytes here to both UTF8 and non-UTF8 outputs